### PR TITLE
Fix sentry-cli not being extracted prior to task execution

### PR DESCRIPTION
**Pull request by markushi**
Fixes https://github.com/getsentry/sentry-android-gradle-plugin/issues/686

Created a base task for all our sentry-cli based tasks. 
During task execution the cli is then extracted if the computed path does not exist.

## :pencil: Checklist

- [ ] I reviewed the submitted code
- [ ] I added tests to verify the changes
- [ ] I updated the docs if needed
- [ ] No breaking changes

### CODE REVIEW

File: plugin-build/src/main/kotlin/io/sentry/android/gradle/SentryCliProvider.kt [hunk 1]
```diff
@@ -93,43 +102,37 @@ internal object SentryCliProvider {
         }
     }
 
-    internal fun searchCliInResources(resourcePath: String): String? {
-        val resourceURL = javaClass.getResource(resourcePath)
-        val resourceFile = resourceURL?.let { File(it.file) }
```
Comment (markushi) [inline plugin-build/src/main/kotlin/io/sentry/android/gradle/SentryCliProvider.kt:98]: I'm really not sure why this was done this way 😅 Shouldn't it be fine to just null check the `resourceURL`?

Comment (romtsn) [reply plugin-build/src/main/kotlin/io/sentry/android/gradle/SentryCliProvider.kt:98]: yeah I also don't know, null check should be fine I guess 👍

File: plugin-build/src/main/kotlin/io/sentry/android/gradle/telemetry/SentryTelemetryService.kt [hunk 2]
```diff
@@ -495,6 +507,9 @@ abstract class SentryCliInfoValueSource : ValueSource<String, InfoParams> {
 
 abstract class SentryCliVersionValueSource : ValueSource<String, VersionParams> {
     interface VersionParams : ValueSourceParameters {
+        @get:Input
+        val projectBuildFolder: Property<File>
```
Comment (romtsn) [inline plugin-build/src/main/kotlin/io/sentry/android/gradle/telemetry/SentryTelemetryService.kt:511]:
```suggestion
        val buildDirectory: Property<File>
```
should we align naming with the one above?

File: plugin-build/src/main/kotlin/io/sentry/android/gradle/tasks/SentryCliExecTask.kt [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+
+    @get:Input
+    @get:Optional
+    abstract val debug: Property<Boolean>
+
+    @get:Input
+    abstract val cliExecutable: Property<String>
+
+    @get:Input
+    abstract val buildDirectory: Property<File>
```
Comment (romtsn) [inline plugin-build/src/main/kotlin/io/sentry/android/gradle/tasks/SentryCliExecTask.kt:26]: `m`: I think you don't even have to pass it as a task input, you can actually access it within the task itself: `project.layout.buildDirectory`

**Review by romtsn — APPROVED** [review]
LGTM, great work!

**Pull request merged**